src/main/java/com/conveyal/gtfs/loader/conditions/FieldInRangeCheck.java

```diff
@@ -0,0 +1,88 @@
+package com.conveyal.gtfs.loader.conditions;
+
+import com.conveyal.gtfs.error.NewGTFSError;
+import com.conveyal.gtfs.loader.Field;
+import com.conveyal.gtfs.loader.LineContext;
+import com.google.common.collect.HashMultimap;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.conveyal.gtfs.error.NewGTFSErrorType.CONDITIONALLY_REQUIRED;
+import static com.conveyal.gtfs.loader.JdbcGtfsLoader.POSTGRES_NULL_TEXT;
+
+/**
+ * Conditional requirement to check that a reference field value is within a defined range and the conditional field
+ * value has not be defined.
+ */
+public class FieldInRangeCheck extends ConditionalRequirement {
+    /** The minimum reference field value if a range check is being performed. */
+    protected int minReferenceValue;
+    /** The maximum reference field value if a range check is being performed. */
+    protected int maxReferenceValue;
+
+    public FieldInRangeCheck(
+        int minReferenceValue,
+        int maxReferenceValue,
+        String dependentFieldName
+    ) {
+        this.minReferenceValue = minReferenceValue;
+        this.maxReferenceValue = maxReferenceValue;
+        this.dependentFieldName = dependentFieldName;
+    }
+
+    /**
+     * If the reference field value is within a defined range and the conditional field value has not be defined, flag
+     * an error.
+     */
+    public Set<NewGTFSError> check(
+        LineContext lineContext,
+        Field referenceField,
+        HashMultimap<String, String> uniqueValuesForFields
+    ) {
+        Set<NewGTFSError> errors = new HashSet<>();
+        String referenceFieldValue = lineContext.getValueForRow(referenceField.name);
+        String conditionalFieldValue = lineContext.getValueForRow(dependentFieldName);
```

Comment:
This same code also appears in `FieldIsEmptyCheck` and `FieldNotEmptyAndMatchesValueCheck`, and to a lesser extent in the other conditions classes. As a result, this could almost be wrapped, so that the `referenceFieldValue` and `conditionalFieldValue` are passed to the `check` function.

Reply:
The `check` method is based on the abstract method in `ConditionalRequirement`. An update would impact conditional checks where `referenceFieldValue` and `conditionalFieldValue` are not required.